### PR TITLE
Feature/us17685 contentful wireup album song

### PR DESCRIPTION
**Pull request by J-Byron**
Responsible for wiring album & song templates up to consume content loaded from contentful

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -33,16 +38,30 @@ include:
 collections_dir: collections
 
 collections:
-  # ...
+  albums: 
+    output: true
+    permalink: /music/:slug
+  songs: 
+    permalink: /music/(:album/:slug)/:slug
+    output: true
```
Comment (seancdavis) [inline _config.yml]: Super nit-picky, let's be consistent with ordering.

File: _config.yml [hunk 2]
```diff
@@ -33,16 +38,30 @@ include:
...
 
 contentful:
   # ...
+  song: 
+    belongs_to:
+      album: songs
```
Comment (seancdavis) [inline _config.yml]: This makes me think you're bringing in all the content from Contentful when we only want albums and songs. You'll want to exclude all the others and add album.

File: _config.yml [hunk 3]
```diff
@@ -33,16 +38,30 @@ include:
...
+      type: albums
+    values:
+      layout: album
+  - scope:
+      path: ""
+      type: songs
     values:
-      js_manifest: application_deferred
+      layout: song
   # ...
```
Comment (seancdavis) [inline _config.yml]: Remove comment.

File: _includes/_logo-block.html [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
       {% include svg/_spotify-logo.svg color="cr-white" class="logo-block-logo" %}
     </a>
-    <a href="https://www.youtube.com/channel/UC5w5QDnJIpeJR_KnLZSEg1Q" target="_blank">
-      {% include svg/_youtube-music-logo.svg color="cr-white" class="logo-block-logo" %}
-    </a>
-    <a href="https://music.apple.com/us/artist/crossroads-music/1277412437" target="_blank">
+      <a href="{{ include.content.youtube_url }}" target="_blank">
+        {% include svg/_youtube-music-logo.svg color="cr-white" class="logo-block-logo" %}
+      </a>
+    <a href="{{ include.content.apple_music_url }}" target="_blank">
```
Comment (seancdavis) [inline _includes/_logo-block.html]: We usually reserve "content" for being more of a body type of field. I'd probably go with something like "links" in this case.

File: _includes/_song-block.html [hunk 5]
```diff
@@ -14,30 +14,30 @@ <h2 class="title">{{ include.title }}</h2>
   <div id="collapse-{{ include.id }}" class="panel-collapse collapse" role="tabpanel">
     <div class="song-block-content">
       <div class="song-block-description">
-        <p>CCLI# 123456789</p>
-        <p> Written By: Austin Livingood, Jorge Mhondera, Philippa Hanna, Robbie Reider</p>
+        <p>CCLI# {{ include.song.ccli_number }}</p>
+        <p> Written By: {{ include.song.written_by }}</p>
```
Comment (seancdavis) [inline _includes/_song-block.html]: Are these required fields? If not, we should account for a zero state.

File: _includes/_song-block.html [hunk 6]
```diff
@@ -14,30 +14,30 @@ <h2 class="title">{{ include.title }}</h2>
...
         <p>All Rights Reserved. Used by Permission.</p>
       </div>
       <div class="download-box">
         <p class="push-bottom">For Worship Leaders</p>
         <p>
-          <a class="link" href="#">
+          <a class="link" href="{{ include.song.lyrics_file }}" download="{{ include.song.slug }}-lyrics.pdf">
```
Comment (seancdavis) [inline _includes/_song-block.html]: 1. Have you tested this? Isn't `lyrics_file` is an asset?
2. What is the download attribute doing?

File: _includes/_song-block.html [hunk 7]
```diff
@@ -14,30 +14,30 @@ <h2 class="title">{{ include.title }}</h2>
...
             <svg class="icon icon-1 text-primary soft-half-right" viewBox="0 0 256 256">
               <use xlink:href="/img/icons/icons.svg#download"></use>
             </svg>
             Download Lyrics
           </a>
         </p>
         <p>
-          <a class="link" href="#">
+          <a class="link" href="{{ include.song.lyrics_file }}" download="{{ include.song.slug }}-chords.pdf">
```
Comment (seancdavis) [inline _includes/_song-block.html]: Same comments here. But also, you're using `lyrics_file` again.

File: _layouts/album.html [hunk 8]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+---
+
+<div class="jumbotron jumbotron-featured" style="background-image: url(/img/jumbotron/you-are-able.png);">
```
Comment (seancdavis) [inline _layouts/album.html]: Background image should come from Contentful.

File: _layouts/album.html [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  </div>
+
+  <div class="page-section">
+    <h4 class="page-section-title">videos</h4>
+
+    {% include _video-player.html video=page.featured_videos.first %}
+
+    <div class="row">
+      {% for featured_video in page.featured_videos %}
+        {% if forloop.index > 1 %}
```
Comment (seancdavis) [inline _layouts/album.html]: That's one way to do it. Another way is to assign another variable to a sliced copy of `page.featured_videos`.

This is fine — just mentioning so you know.

File: _layouts/song.html [hunk 10]
```diff
@@ -3,30 +3,26 @@
 ---
 
 <div class="jumbotron jumbotron-featured" style="background-image: url(/img/jumbotron/jumbotron-song@2x.png); height: 300px;">
-  <a class="jumbotron-nav-link" href="javascript:void(0)">Back to album</a>
+  <a class="jumbotron-nav-link" href="/music/{{page.album.slug }}">Back to album</a>
```
Comment (seancdavis) [inline _layouts/song.html]: I haven't said this everywhere, but let's stay as clean as we can with the code. Indentation looks good, but we want a space inside liquid tags `{{ ... }}` and `{% ... %}` .

File: _layouts/song.html [hunk 11]
```diff
@@ -3,30 +3,26 @@
...
+    {% include _song-block.html song=page %}
   </div>
 
   <div class="page-section">
     <h4 class="page-section-title">lyrics</h4>
     <div class="row">
       <div class="col col-md-8 col-md-offset-2 text-gray-light text-sm">
-        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut magna dui,
+        {{ page.lyrics | markdownify }}
+        <!-- <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut magna dui,
```
Comment (seancdavis) [inline _layouts/song.html]: You can remove the comments now that you've added real content.

File: _layouts/song.html [hunk 12]
```diff
@@ -48,41 +44,48 @@ <h4 class="page-section-title">lyrics</h4>
...
           faucibus congue. Proin vulputate est id accumsan gravida.
-        </p>
+        </p> -->
       </div>
     </div>
   </div>
 
   <div class="page-section">
+
+    {% if page.youtube_url  %}
```
Comment (seancdavis) [inline _layouts/song.html]: I'm not sure I understand the logic here? I thought for song we were using the same logic as album — first video is featured and the others are listed as cards.

File: _layouts/song.html [hunk 13]
```diff
@@ -48,41 +44,48 @@ <h4 class="page-section-title">lyrics</h4>
...
-      <div class="col col-md-4">
-        {% include _video-card.html link="javascript:void(0)" image_url="/img/featured-videos/video-1.jpg" title="Video title" %}
-      </div>
-      <div class="col col-md-4">
-        {% include _video-card.html link="javascript:void(0)" image_url="/img/featured-videos/video-1.jpg" title="Video title" %}
-      </div>
-      <div class="col col-md-4">
-        {% include _video-card.html link="javascript:void(0)" image_url="/img/featured-videos/video-1.jpg" title="Video title" %}
+        <div class="col col-md-8 col-md-offset-2 text-gray-light text-sm">
+          {% include _video-player.html video=page.video %}
```
Comment (seancdavis) [inline _layouts/song.html]: Same here. It's possible I missed some requirements at some point. Let's discuss.

File: music.html [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+---
+
+<main>
+  <div
+    class="jumbotron"
+    style="background-image: url(/img/jumbotron/songs-and-albums.png); height: 250px;"
+  >
+    <div class="container text-white">
+      <h1
+        class="font-family-condensed-extra text-uppercase jumbotron-title text-bottom-left"
```
Comment (seancdavis) [inline music.html]: I know this isn't your issue, but doesn't `.jumbotron-title` remove the need to have these extra classes? May be a Silas question.

File: _layouts/song.html (hunk 10 above)
Comment (seancdavis) [reply _layouts/song.html]: The other thing I noticed here is that you're manually building the url (`href`) when you could call `page.album.url`.

File: _layouts/song.html (hunk 12 above)
Comment (seancdavis) [reply _layouts/song.html]: As we discussed:

- Song doesn't need a YouTube URL (edit the migration)
- The zero-state logic should be wrapped around the entire page section for videos

File: music.html (hunk 14 above)
Comment (seancdavis) [reply music.html]: You can ignore this for now. I believe Jeff is working on it.

File: _includes/_song-block.html (hunk 5 above)
Comment (J-Byron) [reply _includes/_song-block.html]: What did you have in mind as an alternative? Perhaps a default value or exclude the field altogether? I could Also make this a required field in the migration

**Pull request merged**